test/diffrules.jl

```diff
@@ -137,31 +86,28 @@ end
         N, M, K = 64, 5, 10
         q = Hankel.QDHT{1,2}(10, N)
         @testset "Vector" begin
-            A, Ȧ, Ā = ntuple(_ -> randn(rng, N), 3)
-            ȳ = randn(rng)
-            rrule_test(f, ȳ, (A, Ā), (q, nothing))
-            frule_test(f, (A, Ȧ), (q, nothing))
+            A = randn(rng, N)
+            test_rrule(f, A, q ⊢ NoTangent())
+            test_frule(f, A, q ⊢ NoTangent())
         end
 
         @testset "Matrix" begin
             @testset for dim in (1, 2)
                 s = dim == 1 ? (N, M) : (M, N)
                 sy = dim == 1 ? (1, M) : (M, 1)
-                A, Ȧ, Ā = ntuple(_ -> randn(rng, s), 3)
-                ȳ = randn(rng, sy)
-                rrule_test(f, ȳ, (A, Ā), (q, nothing); fkwargs = (dim = dim,))
-                frule_test(f, (A, Ȧ), (q, nothing); fkwargs = (dim = dim,))
+                A = randn(rng, s)
+                test_rrule(f, A, q ⊢ NoTangent(); fkwargs = (dim = dim,))
+                test_frule(f, A, q ⊢ NoTangent(); fkwargs = (dim = dim,))
             end
         end
 
         @testset "Array{<:Real,3}" begin
             s = (M, N, K)
             sy = (M, 1, K)
             dim = 2
-            A, Ȧ, Ā = ntuple(_ -> randn(rng, s), 3)
-            ȳ = randn(rng, sy)
-            rrule_test(f, ȳ, (A, Ā), (q, nothing); fkwargs = (dim = dim,))
-            frule_test(f, (A, Ȧ), (q, nothing); fkwargs = (dim = dim,))
+            A = randn(rng, s)
+            test_rrule(f, A, q ⊢ NoTangent(); fkwargs = (dim = dim,), check_inferred=false)
```

Comment:
This is the only thing I would flag manually, it looks like the inference is broken for one of the pullbacks. It could be that this PR broke it - but I imagine we just didn't test for this in the earlier versions of ChainRulesTestUtils?

@sethaxen this is ready for your 👀

Reply:
I think we just didn't test
https://github.com/JuliaDiff/ChainRulesTestUtils.jl/blob/v0.4.3/src/testers.jl

Reply:
Oh, great that we're testing that now. It was an easy enough fix for type inference.